controllers/release/adapter.go

```diff
@@ -591,3 +717,31 @@ func (a *adapter) validatePipelineRef() *controller.ValidationResult {
 
 	return &controller.ValidationResult{Valid: true}
 }
+
+// validateSinglePipeline checks that the Pipeline is defined exclusively in the ReleasePlan or in the ReleasePlanAdmission.
+func (a *adapter) validateSinglePipeline() *controller.ValidationResult {
+	releasePlan, err := a.loader.GetReleasePlan(a.ctx, a.client, a.release)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			a.release.MarkValidationFailed(err.Error())
+			return &controller.ValidationResult{Valid: false}
+		}
+
+		return &controller.ValidationResult{Err: err}
+	}
+
+	releasePlanAdmission, err := a.loader.GetActiveReleasePlanAdmissionFromRelease(a.ctx, a.client, a.release)
+	if err != nil && errors.IsNotFound(err) {
+		a.release.MarkValidationFailed(err.Error())
+		return &controller.ValidationResult{Valid: false}
+	}
+
+	if releasePlanAdmission != nil {
+		if releasePlan.Spec.Pipeline != nil && releasePlanAdmission.Spec.Pipeline != nil {
+			a.release.MarkValidationFailed("pipeline should be set only in the ReleasePlan or in the ReleasePlanAdmission")
```

Comment:
We actually don't care about this. I think that the proper check, provided that we want to support both in the future, is to check that Target and Pipeline are not defined at the same time in the RP. No need to check the RPA.

Reply:
Ok. This makes it easier.

Reply:
I actually disagree with this and think it is a much better check that a pipeline is not defined in both the RP and RPA. Regardless, we all agree this is temporary code (disallowing a pipeline in both spots) so it doesn't matter to me - do whatever is easiest, no need to waste cycles on temporary things

Reply:
I ended up changing it to not use the RPA.